tomviz/operators/EditOperatorDialog.cxx

```diff
@@ -165,14 +168,25 @@ void EditOperatorDialog::onApply()
     }
   }
   if (this->Internals->needsToBeAdded) {
-    this->Internals->dataSource->addOperator(this->Internals->Op);
+    this->Internals->dataSource->pipeline()->resume(
+      this->Internals->dataSource);
     this->Internals->needsToBeAdded = false;
   }
 }
 
 void EditOperatorDialog::onClose()
 {
   this->Internals->saveGeometry(this->geometry());
+  if (this->Internals->needsToBeAdded) {
+    // Since for now operators can't be programmatically removed
+    // (i.e. all removals are assumed to be initialized
+    // from the gui in PipelineModel), we need to do a workaround and have the
+    // ModuleManaged emit a signal that is captured by the PipelineModel,
+    // which eventually will lead to the removal of the operator.
+    ModuleManager::instance().removeOperator(this->Internals->Op);
+    this->Internals->dataSource->pipeline()->resume(
+      this->Internals->dataSource);
```

Comment:
As we are removing the operator we don't want to execute the pipeline, so we should do ```resume(false)```